core/v2/core_test.go

```diff
@@ -0,0 +1,267 @@
+		2: int(operatorCount),
+	})
+	assert.NoError(t, err)
+
+	blobsMap := make([]map[core.QuorumID]map[core.OperatorID][]*encoding.Frame, 0, len(headers))
+
+	for z, header := range headers {
+
+		blob := blobs[z]
+
+		params, err := header.GetEncodingParams()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		chunks, err := p.GetFrames(blob, params)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		state, err := cst.GetOperatorState(context.Background(), uint(header.ReferenceBlockNumber), header.QuorumNumbers)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		blobMap := make(map[core.QuorumID]map[core.OperatorID][]*encoding.Frame)
+
+		for _, quorum := range header.QuorumNumbers {
+
+			assignments, err := corev2.GetAssignments(state, header.Version, quorum)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			blobMap[quorum] = make(map[core.OperatorID][]*encoding.Frame)
+
+			for opID, assignment := range assignments {
+
+				blobMap[quorum][opID] = chunks[assignment.StartIndex : assignment.StartIndex+assignment.NumChunks]
+
+			}
+
+		}
+
+		blobsMap = append(blobsMap, blobMap)
+	}
+
+	// Invert the blobsMap
+	inverseMap := make(map[core.OperatorID][]*corev2.BlobShard)
+	for blobIndex, blobMap := range blobsMap {
+		for quorum, operatorMap := range blobMap {
+			for operatorID, frames := range operatorMap {
+
+				if _, ok := inverseMap[operatorID]; !ok {
+					inverseMap[operatorID] = make([]*corev2.BlobShard, 0)
+				}
+				if len(inverseMap[operatorID]) < blobIndex+1 {
+					inverseMap[operatorID] = append(inverseMap[operatorID], &corev2.BlobShard{
+						BlobCertificate: headers[blobIndex],
+						Chunks:          make(map[core.QuorumID][]*encoding.Frame),
+					})
+				}
+				if len(frames) == 0 {
+					continue
+				}
+				inverseMap[operatorID][blobIndex].Chunks[quorum] = append(inverseMap[operatorID][blobIndex].Chunks[quorum], frames...)
+
+			}
+		}
+	}
+
+	return inverseMap, cst
+
+}
+
+// checkBatchByUniversalVerifier runs the verification logic for each DA node in the current OperatorState, and returns an error if any of
+// the DA nodes' validation checks fails
+func checkBatchByUniversalVerifier(cst core.IndexedChainState, packagedBlobs map[core.OperatorID][]*corev2.BlobShard, pool common.WorkerPool) error {
+
+	ctx := context.Background()
+
+	quorums := []core.QuorumID{0, 1}
+	state, _ := cst.GetIndexedOperatorState(context.Background(), 0, quorums)
+	// numBlob := len(encodedBlobs)
+
+	var errList *multierror.Error
+
+	for id := range state.IndexedOperators {
+
+		val := corev2.NewShardValidator(v, cst, id)
+
+		blobs := packagedBlobs[id]
+
+		err := val.ValidateBlobs(ctx, blobs, pool)
+		if err != nil {
+			errList = multierror.Append(errList, err)
+		}
+	}
+
+	return errList.ErrorOrNil()
+
+}
+
+func TestValidationSucceeds(t *testing.T) {
+
+	// operatorCounts := []uint{1, 2, 4, 10, 30}
+
+	// numBlob := 3 // must be greater than 0
+	// blobLengths := []int{1, 32, 128}
+
+	operatorCounts := []uint{4}
+
+	numBlob := 1 // must be greater than 0
```

Comment:
comment from @hopeyen: could we add more tests such as different number of blobs and blobs that would be considered invalid by the verifier?